File: client-app/src/types/profession.ts
export interface Resource {
    id: string;
    url: string;
    isAffiliate: boolean;
}

export interface RoadmapStep {
    id: string;
    title: string;
    description: string;
    orderIndex: number;
    mustKnow: boolean;
    resources: Resource[];
}

export interface Profession {
    id: string;
    title: string;
    slug: string;
    avgSalary: number;
    difficultyLevel: string;
    roadmapSteps: RoadmapStep[];
}

File: src/CareerPath.Core/Entities/Profession.cs
using System.ComponentModel.DataAnnotations;

namespace CareerPath.Core.Entities;

public class Profession
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(100)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [MaxLength(150)]
    public string Slug { get; set; } = string.Empty;

    public decimal AvgSalary { get; set; }

    [MaxLength(50)]
    public string DifficultyLevel { get; set; } = string.Empty; // Easy, Medium, Hard

    // Navigation Property
    public ICollection<RoadmapStep> RoadmapSteps { get; set; } = new List<RoadmapStep>();
}
